### PR TITLE
Switch to slack notifications

### PR DESCRIPTION
**Pull request by bahamat**
What I did:

* cloned `joyMattermostNotification.groovy` to `joySlackNotifications.groovy`
* Changed references in `joySlackNotifications.groovy` from MM to Slack (reference: <https://github.com/jenkinsci/slack-plugin/blob/slack-2.48/src/main/resources/jenkins/plugins/slack/workflow/SlackSendStep/help-channel.html>)
* ~~Made `joyMattermostNotification.groovy` a symlink to `joySlackNotifications.groovy` so that we don't have to go change all the other repos (according to hound, there are 95 repos that would need updating).~~
* Deleted `joyMattermostNotification.groovy` since all repos consuming this plugin are supposed to point to a specific git tag.

After this PR is merged I'll tag the squashed commit so we can update all other repos. (See TOOLS-2521)

### CODE REVIEW

**Comment by bahamat** [comment]
Not having MM anymore, seems to have broken at least *some* builds (e.g., node-vmamd), but I don't know *why*, and I don't know why most builds don't seem to have been broken.

**Comment by bahamat** [comment]
References to `joyMattermostNotification`:

<https://hound.joyent.us/?q=joyMattermostNotification&i=nope&files=&repos=>

File: vars/joySlackNotifications.groovy [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
- * Notify the jenkins mattermost channel, only on master/release branches, and
+ * Notify the jenkins Slack channel, only on master/release branches, and
  * don't neglect colors or emoji.
 */
 void call(Map args = [:]) {
     String channel = args.channel ?: 'jenkins';
     String comment = args.comment ?: '';
 
-    // addapted from https://github.com/jenkinsci/mattermost-plugin/blob/mattermost-2.7.0/src/main/java/jenkins/plugins/mattermost/ActiveNotifier.java
+    // addapted from https://github.com/jenkinsci/slack-plugin/blob/slack-2.48/src/main/java/jenkins/plugins/slack/ActiveNotifier.java
```
Comment (danmcd) [inline vars/joySlackNotifications.groovy]: "adapted"

**Review by danmcd — COMMENTED** [review]
Spelling nit is all.  You mentioned fixing copyright too, so fix 'em both and you'll get an easy +1.

File: vars/joySlackNotifications.groovy (hunk 1 above)
Comment (bahamat) [reply vars/joySlackNotifications.groovy]: Ha. That’s just what was there before, but I’ll fix that too :-)

**Review by danmcd — APPROVED** [review]
Thanks for the two nit fixes.  Ship it.

**Pull request merged**